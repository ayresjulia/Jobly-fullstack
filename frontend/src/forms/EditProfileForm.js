import React, { useState } from "react";
import { Form, FormGroup, Label, Input, Button } from "reactstrap";
import { useHistory } from "react-router-dom";
import "./EditProfileForm.css";
import JoblyApi from "../Api";
import Alert from "../Alert";

/** Form to edit user info in db. */

const EditProfileForm = ({ currentUser }) => {
	const INITIAL_STATE = {
		username: currentUser.username,
		firstName: currentUser.firstName,
		lastName: currentUser.lastName,
		email: currentUser.email,
		password: ""
	};
	const [ formData, setFormData ] = useState(INITIAL_STATE);
	const [ formErrors, setFormErrors ] = useState([]);
	const [ saveConfirmed, setSaveConfirmed ] = useState(false);

	const history = useHistory();

	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((formData) => ({
			...formData,
			[name]: value
		}));
	};

	/** Submit form and add new user data to db. */

	const handleSubmit = async (e) => {
		e.preventDefault();
		let profileData = {
			firstName: formData.firstName,
			lastName: formData.lastName,
			email: formData.email,
			password: formData.password
		};
		try {
			let username = currentUser.username;
			let updatedUser = await JoblyApi.saveProfile(username, profileData);
			console.log(updatedUser);
		} catch (e) {
			setFormErrors(e);
			return;
		}
		setFormData((f) => ({ ...f, password: "" }));
		setFormErrors([]);
		setSaveConfirmed(true);

		history.push("/");
	};

	return (
		<div className="Form">
			<Label htmlFor="username">Username</Label>
			<p>
				<b>{currentUser.username}</b>
			</p>
			<Form className="Form-body" onSubmit={handleSubmit}>
				<FormGroup>
					<Label htmlFor="firstName">First Name</Label>
					<Input
						className="Form-input"
						name="firstName"
						id="firstName"
						value={formData.firstName}
						onChange={handleChange}
					/>
				</FormGroup>
				<FormGroup>
					<Label htmlFor="lastName">Last Name</Label>
					<Input
						className="Form-input"
						name="lastName"
						id="lastName"
						value={formData.lastName}
						onChange={handleChange}
					/>
				</FormGroup>
				<FormGroup>
					<Label htmlFor="email">Email</Label>
					<Input
						className="Form-input"
						name="email"
						id="email"
						value={formData.email}
						onChange={handleChange}
					/>
				</FormGroup>
				<FormGroup>
					<Label htmlFor="password">Confirm password to make changes</Label>
					<Input
						className="Form-input"
						type="password"
						name="password"
						id="password"
						value={formData.password}
						onChange={handleChange}
					/>
				</FormGroup>
				{formErrors.length ? <Alert type="danger" messages={formErrors} /> : null}

				{saveConfirmed ? (
					<Alert type="success" messages={[ "Updated successfully." ]} />
				) : null}
				<Button className="Form-btn btn-success">Save</Button>
			</Form>
		</div>
	);
};

export default EditProfileForm;
