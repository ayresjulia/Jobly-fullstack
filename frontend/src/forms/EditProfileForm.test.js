import { render } from "@testing-library/react";
import React from "react";
import EditProfileForm from "./EditProfileForm";

test("it renders without crashing", () => {
	let username = "testuser";
	render(<EditProfileForm currentUser={username} />);
});

it("matches snapshot", () => {
	let username = "testuser";
	const { asFragment } = render(<EditProfileForm currentUser={username} />);
	expect(asFragment()).toMatchSnapshot();
});

it("renders a field in the form", () => {
	let username = "testuser";
	const { getByText } = render(<EditProfileForm currentUser={username} />);
	expect(getByText("Confirm password to make changes")).toBeInTheDocument();
});
