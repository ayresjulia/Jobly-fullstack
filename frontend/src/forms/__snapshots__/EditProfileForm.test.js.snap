// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Form"
  >
    <label
      class=""
      for="username"
    >
      Username
    </label>
    <p>
      <b />
    </p>
    <form
      class="Form-body"
    >
      <div
        class="form-group"
      >
        <label
          class=""
          for="firstName"
        >
          First Name
        </label>
        <input
          class="Form-input form-control"
          id="firstName"
          name="firstName"
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class=""
          for="lastName"
        >
          Last Name
        </label>
        <input
          class="Form-input form-control"
          id="lastName"
          name="lastName"
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class=""
          for="email"
        >
          Email
        </label>
        <input
          class="Form-input form-control"
          id="email"
          name="email"
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class=""
          for="password"
        >
          Confirm password to make changes
        </label>
        <input
          class="Form-input form-control"
          id="password"
          name="password"
          type="password"
          value=""
        />
      </div>
      <button
        class="Form-btn btn-success btn btn-secondary"
      >
        Save
      </button>
    </form>
  </div>
</DocumentFragment>
`;
